RadClass/RadClass.py

```diff
@@ -171,17 +200,17 @@ def iterate(self):
         Only runs for a set node (datapath) with data already queued.
         '''
         bar = progressbar.ProgressBar(max_value=100, redirect_stdout=True)
-        inverse_dt = 1.0 / (self.processor.timestamps[-1]
-                            - self.processor.timestamps[0])
+        inverse_dt = 1.0 / (self.stop_time - self.start_time)
 
         log_interval = 10000  # number of samples analyzed between log updates
         running = True  # tracks whether to end analysis
         while running:
             # print status at set intervals
-            if np.where(self.processor.timestamps == self.working_time)[0][0] % log_interval == 0:
-                bar.update(round((self.working_time - self.processor.timestamps[0]) * inverse_dt, 4)*100)
+            current_time = self.processor.timestamps[self.current_i]
+            if (current_time - self.start_time) % log_interval == 0:
```

Comment:
I know that the index equates to seconds, but this might be cleaner if you used `current_i` and `start_i` since `log_interval` is really in index space?

Reply:
Agreed, but should I make the same change to line 211:

```
bar.update(round((current_time - self.start_time) * inverse_dt, 4)*100)
```

Which would then probably mean changing `inverse_dt` as well?

```
inverse_dt = 1.0 / (self.stop_time - self.start_time)
```

Reply:
Yes, or conversely, we can interpret `log_interval` to be in seconds and leave it all.

I keep leaving open the possibility that these methods could be used for a dataset with a different time between data points